src/protocol.cpp

```diff
@@ -87,25 +87,16 @@ const static std::string allNetMessageTypes[] = {
 };
 const static std::vector<std::string> allNetMessageTypesVec(std::begin(allNetMessageTypes), std::end(allNetMessageTypes));
 
-CMessageHeader::CMessageHeader()
-{
-    memset(pchMessageStart, 0, MESSAGE_START_SIZE);
-    memset(pchCommand, 0, sizeof(pchCommand));
-    memset(pchChecksum, 0, CHECKSUM_SIZE);
-}
-
 CMessageHeader::CMessageHeader(const MessageStartChars& pchMessageStartIn, const char* pszCommand, unsigned int nMessageSizeIn)
```

Comment:
We could also change the type here to `std::string` or `std::span` instead of a loose pointer, at some point and clean up the loop (oh and rename the argument to `message_type` maybe instead of command, when we're at it). No need to do so in this PR though.